local/snapshot.go

```diff
@@ -180,16 +174,13 @@ func (ln *localNetwork) SaveSnapshot(ctx context.Context, snapshotName string) (
 		SubnetConfigFiles:  ln.subnetConfigFiles,
 	}
 
-	for _, nodeConfig := range nodesConfig {
-		// no need to save this, will be generated automatically on snapshot load
-		networkConfig.NodeConfigs = append(networkConfig.NodeConfigs, nodeConfig)
-	}
+	// no need to save this, will be generated automatically on snapshot load
+	networkConfig.NodeConfigs = append(networkConfig.NodeConfigs, maps.Values(nodesConfig)...)
```

Comment:
why not just nodesConfig.... instead of maps.Values(nodesConfig)... ? I think the original copies the same object?

Reply:
This is functionally equivalent to what was here before. We can't append `nodesConfig` because it's a `map[string]node.Config`, not a `node.Config`. If you look inside the implementation of `maps.Values` you'll see it's doing the exact same thing:

```go
// Values returns the values of the map m.
// The values will be in an indeterminate order.
func Values[M ~map[K]V, K comparable, V any](m M) []V {
	r := make([]V, 0, len(m))
	for _, v := range m {
		r = append(r, v)
	}
	return r
}
```